--- a/libc/str/tprecode8to16.c
+++ b/libc/str/tprecode8to16.c
@@ -16,12 +16,14 @@
 │ TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR             │
 │ PERFORMANCE OF THIS SOFTWARE.                                                │
 ╚─────────────────────────────────────────────────────────────────────────────*/
-#include <arm_neon.h>
 #include <stdint.h>
 #include <string.h>
 #include "libc/dce.h"
 #include "libc/str/str.h"
 #include "libc/str/thompike.h"
 #include "libc/str/utf16.h"
+#if defined(__aarch64__)
+#include <arm_neon.h>
 #include "third_party/aarch64/arm_neon.internal.h"
+#endif
 #include "libc/isystem/emmintrin.h"
