diff --git a/libc/intrin/sigblock.c b/libc/intrin/sigblock.c
index 8856d028..9424d86b 100644
--- a/libc/intrin/sigblock.c
+++ b/libc/intrin/sigblock.c
@@ -24,8 +24,6 @@
 // that always need to block signals we avoid the distraction of their
 // ftrace and strace output being muddied with sigprocmask lines. it's
 // usually better that sigprocmask only strace the user is calling it.
 // plus, since we have a very specific use case, this code goes faster
-
-struct Signals __sig;
 
 sigset_t __sig_block(void) {
   if (IsWindows() || IsMetal()) {
