diff -ruN a/libc/integral/normalize.inc b/libc/integral/normalize.inc
--- a/libc/integral/normalize.inc
+++ b/libc/integral/normalize.inc
@@ -53,9 +53,16 @@
 #ifndef __has_builtin
 #define __has_builtin(x) 0
 #endif
 #ifndef __has_cpp_attribute
 #define __has_cpp_attribute(x) 0
 #endif
+#if (defined(__x86_64__) || defined(__i386__)) && !defined(COSMO_DISABLE_MMX)
+#if defined(__has_builtin)
+#if !__has_builtin(__builtin_ia32_vec_init_v2si)
+#define COSMO_DISABLE_MMX 1
+#endif
+#endif
+#endif
 #ifndef __has_extension
 #define __has_extension(x) 0
 #endif
