diff -u a/libc/intrin/demangle.c b/libc/intrin/demangle.c
--- a/libc/intrin/demangle.c
+++ b/libc/intrin/demangle.c
@@ -30,15 +30,15 @@
 #include <stdalign.h>
 #include <stdbool.h>
 #include <stddef.h>
 #include <stdint.h>
 #include "libc/runtime/runtime.h"
 
-#if defined(__aarch64__)
-#define DEMANGLE_JMPBUF jmp_buf
-#define DEMANGLE_SETJMP(env) setjmp(env)
-#define DEMANGLE_LONGJMP(env, val) longjmp(env, val)
-#else
-#define DEMANGLE_JMPBUF intptr_t[5]
-#define DEMANGLE_SETJMP(env) __builtin_setjmp(env)
-#define DEMANGLE_LONGJMP(env, val) __builtin_longjmp(env, val)
-#endif
+typedef jmp_buf demangle_jmpbuf;
+#if defined(__aarch64__)
+#define DEMANGLE_SETJMP(env) setjmp(env)
+#define DEMANGLE_LONGJMP(env, val) longjmp(env, val)
+#else
+#define DEMANGLE_SETJMP(env) __builtin_setjmp(env)
+#define DEMANGLE_LONGJMP(env, val) __builtin_longjmp(env, val)
+#endif
+#define DEMANGLE_JMPBUF demangle_jmpbuf
