diff -u a/libc/str/tprecode8to16.c b/libc/str/tprecode8to16.c
--- a/libc/str/tprecode8to16.c
+++ b/libc/str/tprecode8to16.c
@@ -21,11 +21,13 @@
 #include "libc/str/str.h"
 #include "libc/str/thompike.h"
 #include "libc/str/utf16.h"
 #if defined(__aarch64__)
 #include <arm_neon.h>
 #include "third_party/aarch64/arm_neon.internal.h"
 #endif
-#include "libc/isystem/emmintrin.h"
+#if defined(__x86_64__) || defined(__i386__)
+#include "libc/isystem/emmintrin.h"
+#endif
 
 #if !IsModeDbg()
 #if defined(__x86_64__)
diff -u a/libc/str/tprecode16to8.c b/libc/str/tprecode16to8.c
--- a/libc/str/tprecode16to8.c
+++ b/libc/str/tprecode16to8.c
@@ -19,9 +19,13 @@
 #include "libc/dce.h"
 #include "libc/fmt/conv.h"
 #include "libc/str/str.h"
 #include "libc/str/utf16.h"
-#include "third_party/aarch64/arm_neon.internal.h"
-#include "libc/isystem/emmintrin.h"
+#if defined(__aarch64__)
+#include "third_party/aarch64/arm_neon.internal.h"
+#endif
+#if defined(__x86_64__) || defined(__i386__)
+#include "libc/isystem/emmintrin.h"
+#endif
 
 #if !IsModeDbg()
 #if defined(__x86_64__)
diff -u a/libc/str/strcasestr.c b/libc/str/strcasestr.c
--- a/libc/str/strcasestr.c
+++ b/libc/str/strcasestr.c
@@ -21,9 +21,13 @@
 #include "libc/ctype.h"
 #include "libc/mem/alloca.h"
 #include "libc/runtime/stack.h"
 #include "libc/str/tab.h"
-#include "third_party/aarch64/arm_neon.internal.h"
-#include "libc/isystem/immintrin.h"
+#if defined(__aarch64__) && defined(__ARM_NEON)
+#include "third_party/aarch64/arm_neon.internal.h"
+#endif
+#if defined(__x86_64__) || defined(__i386__)
+#include "libc/isystem/immintrin.h"
+#endif
 
 static int ToUpper(int c) {
   return 'a' <= c && c <= 'z' ? c - ('a' - 'A') : c;
diff -u a/libc/str/strstr.c b/libc/str/strstr.c
--- a/libc/str/strstr.c
+++ b/libc/str/strstr.c
@@ -17,9 +17,13 @@
 │ PERFORMANCE OF THIS SOFTWARE.                                                │
 ╚─────────────────────────────────────────────────────────────────────────────*/
 #include "libc/str/str.h"
 #include "libc/str/kmp.h"
-#include "third_party/aarch64/arm_neon.internal.h"
-#include "libc/isystem/immintrin.h"
+#if defined(__aarch64__) && defined(__ARM_NEON)
+#include "third_party/aarch64/arm_neon.internal.h"
+#endif
+#if defined(__x86_64__) || defined(__i386__)
+#include "libc/isystem/immintrin.h"
+#endif
 
 /**
  * Searches for substring.
diff -u a/libc/str/memmem.c b/libc/str/memmem.c
--- a/libc/str/memmem.c
+++ b/libc/str/memmem.c
@@ -19,8 +19,12 @@
 #include "libc/intrin/likely.h"
 #include "libc/str/kmp.h"
 #include "libc/str/str.h"
-#include "third_party/aarch64/arm_neon.internal.h"
-#include "libc/isystem/emmintrin.h"
+#if defined(__aarch64__) && defined(__ARM_NEON)
+#include "third_party/aarch64/arm_neon.internal.h"
+#endif
+#if defined(__x86_64__) || defined(__i386__)
+#include "libc/isystem/emmintrin.h"
+#endif
 
 /**
  * Searches for fixed-length substring in memory region.
