diff -u a/libc/thread/posixthread.internal.h b/libc/thread/posixthread.internal.h
--- a/libc/thread/posixthread.internal.h
+++ b/libc/thread/posixthread.internal.h
@@ -68,7 +68,7 @@ struct PosixThread {
   struct _pthread_cleanup_buffer *pt_cleanup;
   _Atomic(uint64_t) pt_blkmask;
   int64_t pt_event;
   locale_t pt_locale;
-  intptr_t pt_exiter[5];
+  jmp_buf pt_exiter;
   pthread_attr_t pt_attr;
   atomic_bool pt_intoff;
 };
diff -u a/libc/thread/pthread_create.c b/libc/thread/pthread_create.c
--- a/libc/thread/pthread_create.c
+++ b/libc/thread/pthread_create.c
@@ -169,7 +169,7 @@ static int PosixThread(void *arg) {
     unassert(!sigaltstack(ss, 0));
   }
 
   // set long jump handler so pthread_exit can bring control back here
-  if (!__builtin_setjmp(pt->pt_exiter)) {
+  if (!setjmp(pt->pt_exiter)) {
     // setup signals for new thread
     pt->pt_attr.__sigmask &= ~(1ull << (SIGTHR - 1));
     if (IsWindows() || IsMetal()) {
diff -u a/libc/thread/pthread_getname_np.c b/libc/thread/pthread_getname_np.c
--- a/libc/thread/pthread_getname_np.c
+++ b/libc/thread/pthread_getname_np.c
@@ -78,7 +78,8 @@ errno_t pthread_getname_impl(struct PosixThread *pt, char *name, size_t size) {
     }
     return 0;
 
-  } else if (IsNetbsd() || IsOpenbsd()) {
+  } else if (IsNetbsd() || IsOpenbsd()) {
+#if defined(__x86_64__)
     int ax;
     char cf;
     long dx, si;
@@ -100,6 +101,9 @@ errno_t pthread_getname_impl(struct PosixThread *pt, char *name, size_t size) {
       }
     } else {
       return ax;
     }
+#else
+    return ENOSYS;
+#endif
 
   } else {
     return ENOSYS;
