diff -u a/libc/stdio/rdseed.c b/libc/stdio/rdseed.c
--- a/libc/stdio/rdseed.c
+++ b/libc/stdio/rdseed.c
@@ -16,8 +16,10 @@
 │ TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR             │
 │ PERFORMANCE OF THIS SOFTWARE.                                                │
 ╚─────────────────────────────────────────────────────────────────────────────*/
+#if defined(__x86_64__) || defined(__i386__)
 #include "libc/intrin/asmflag.h"
 #include "libc/nexgen32e/x86feature.h"
+#endif
 #include "libc/stdio/rand.h"
 #include "libc/stdio/stdio.h"
 #include "libc/sysv/consts/grnd.h"
@@ -42,6 +44,7 @@
   int i;
   char cf;
   uint64_t x;
+#if defined(__x86_64__) || defined(__i386__)
   if (X86_HAVE(RDSEED)) {
     for (i = 0; i < 10; ++i) {
       asm volatile(CFLAG_ASM("rdseed\t%1")
@@ -54,4 +57,4 @@
     }
   }
   return rdrand();
-}
+}
