--- a/ape/aarch64.lds	2025-01-05 17:05:49
+++ b/ape/aarch64.lds	2026-01-05 09:26:10
@@ -231,38 +231,38 @@
 
   . = DATA_SEGMENT_END(.);
 
-  .stab               0 : { *(.stab) }
-  .stabstr            0 : { *(.stabstr) }
-  .stab.excl          0 : { *(.stab.excl) }
-  .stab.exclstr       0 : { *(.stab.exclstr) }
-  .stab.index         0 : { *(.stab.index) }
-  .stab.indexstr      0 : { *(.stab.indexstr) }
-  .gnu.build.attributes : { *(.gnu.build.attributes .gnu.build.attributes.*) }
-  .debug              0 : { *(.debug) }
-  .line               0 : { *(.line) }
-  .debug_srcinfo      0 : { *(.debug_srcinfo) }
-  .debug_sfnames      0 : { *(.debug_sfnames) }
-  .debug_aranges      0 : { *(.debug_aranges) }
-  .debug_pubnames     0 : { *(.debug_pubnames) }
-  .debug_info         0 : { *(.debug_info .gnu.linkonce.wi.*) }
-  .debug_abbrev       0 : { *(.debug_abbrev) }
-  .debug_line         0 : { *(.debug_line .debug_line.* .debug_line_end) }
-  .debug_frame        0 : { *(.debug_frame) }
-  .debug_str          0 : { *(.debug_str) }
-  .debug_loc          0 : { *(.debug_loc) }
-  .debug_macinfo      0 : { *(.debug_macinfo) }
-  .debug_weaknames    0 : { *(.debug_weaknames) }
-  .debug_funcnames    0 : { *(.debug_funcnames) }
-  .debug_typenames    0 : { *(.debug_typenames) }
-  .debug_varnames     0 : { *(.debug_varnames) }
-  .debug_pubtypes     0 : { *(.debug_pubtypes) }
-  .debug_ranges       0 : { *(.debug_ranges) }
-  .debug_macro        0 : { *(.debug_macro) }
-  .debug_addr         0 : { *(.debug_addr) }
-  .ARM.attributes     0 : { KEEP(*(.ARM.attributes)) KEEP(*(.gnu.attributes)) }
-  .note.gnu.arm.ident 0 : { KEEP(*(.note.gnu.arm.ident)) }
+  .stab : { *(.stab) }
+  .stabstr : { *(.stabstr) }
+  .stab.excl : { *(.stab.excl) }
+  .stab.exclstr : { *(.stab.exclstr) }
+  .stab.index : { *(.stab.index) }
+  .stab.indexstr : { *(.stab.indexstr) }
+  .gnu.build.attributes : { *(.gnu.build.attributes .gnu.build.attributes.*) }
+  .debug : { *(.debug) }
+  .line : { *(.line) }
+  .debug_srcinfo : { *(.debug_srcinfo) }
+  .debug_sfnames : { *(.debug_sfnames) }
+  .debug_aranges : { *(.debug_aranges) }
+  .debug_pubnames : { *(.debug_pubnames) }
+  .debug_info : { *(.debug_info .gnu.linkonce.wi.*) }
+  .debug_abbrev : { *(.debug_abbrev) }
+  .debug_line : { *(.debug_line .debug_line.* .debug_line_end) }
+  .debug_frame : { *(.debug_frame) }
+  .debug_str : { *(.debug_str) }
+  .debug_loc : { *(.debug_loc) }
+  .debug_macinfo : { *(.debug_macinfo) }
+  .debug_weaknames : { *(.debug_weaknames) }
+  .debug_funcnames : { *(.debug_funcnames) }
+  .debug_typenames : { *(.debug_typenames) }
+  .debug_varnames : { *(.debug_varnames) }
+  .debug_pubtypes : { *(.debug_pubtypes) }
+  .debug_ranges : { *(.debug_ranges) }
+  .debug_macro : { *(.debug_macro) }
+  .debug_addr : { *(.debug_addr) }
+  .ARM.attributes : { KEEP(*(.ARM.attributes)) KEEP(*(.gnu.attributes)) }
+  .note.gnu.arm.ident : { KEEP(*(.note.gnu.arm.ident)) }
 
-  .zip 0 : {
+  .zip : {
     KEEP(*(.zip.file))
     __zip_cdir_start = .;
     KEEP(*(.zip.cdir))
@@ -282,6 +282,7 @@
     *(.yoink)
     *(.head)
   }
+  . = MAX(., _end);
 }
 
 ape_stack_vaddr = DEFINED(ape_stack_vaddr) ? ape_stack_vaddr : 0x700000000000;
