diff --git a/BUILD.bazel b/BUILD.bazel
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -35,6 +35,12 @@ expand_template(
     template = "gzread_mangle.h.in",
 )
 
+config_setting(
+    name = "is_windows_clang_mingw",
+    constraint_values = ["@platforms//os:windows"],
+    flag_values = {"@rules_cc//cc/compiler:compiler": "clang"},
+)
+
 cc_library(
     name = "zlib-ng",
     srcs = [
@@ -116,6 +122,12 @@ cc_library(
         ":zlib_gen",
     ],
     copts = select({
+        ":is_windows_clang_mingw": [
+            "-std=c11",
+            "-Wno-deprecated-non-prototype",
+            "-Wno-unused-variable",
+            "-Wno-implicit-function-declaration",
+        ],
         "@platforms//os:windows": [
             "/wd4127",  # conditional expression is constant
             "/wd4131",  # old-style declarator
