diff --git a/utils/bazel/llvm-project-overlay/clang/BUILD.bazel b/utils/bazel/llvm-project-overlay/clang/BUILD.bazel
index ac4e3813fbba..1e03e59e1dcf 100644
--- a/utils/bazel/llvm-project-overlay/clang/BUILD.bazel
+++ b/utils/bazel/llvm-project-overlay/clang/BUILD.bazel
@@ -2,6 +2,8 @@
 # See https://llvm.org/LICENSE.txt for license information.
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
+load("@bazel_skylib//rules/directory:directory.bzl", "directory")
+load("@bazel_skylib//rules/directory:subdirectory.bzl", "subdirectory")
 load("@rules_python//python:defs.bzl", "py_binary")
 load(
     "//:vars.bzl",
@@ -1656,6 +1658,24 @@ genrule(
     toolchains = [":workspace_root"],
 )
 
+filegroup(
+    name = "builtin_headers_files",
+    srcs = builtin_headers,
+)
+
+directory(
+    name = "builtin_headers_dir",
+    srcs = [":builtin_headers_files"],
+    visibility = ["//visibility:public"],
+)
+
+subdirectory(
+    name = "builtin_headers_include_directory",
+    path = "lib/clang/21/include",
+    parent = ":builtin_headers_dir",
+    visibility = ["//visibility:public"],
+)
+
 cc_library(
     name = "frontend",
     srcs = glob([
diff --git a/utils/bazel/llvm-project-overlay/llvm/config.bzl b/utils/bazel/llvm-project-overlay/llvm/config.bzl
index 2309175d04a9..4b3c6b2791db 100644
--- a/utils/bazel/llvm-project-overlay/llvm/config.bzl
+++ b/utils/bazel/llvm-project-overlay/llvm/config.bzl
@@ -28,8 +28,8 @@ def native_arch_defines(arch, triple):
 
 posix_defines = [
     "LLVM_ON_UNIX=1",
-    "HAVE_BACKTRACE=1",
-    "BACKTRACE_HEADER=<execinfo.h>",
+    # "HAVE_BACKTRACE=1",
+    # "BACKTRACE_HEADER=<execinfo.h>",
     r'LTDL_SHLIB_EXT=\".so\"',
     r'LLVM_PLUGIN_EXT=\".so\"',
     "LLVM_ENABLE_LLVM_EXPORT_ANNOTATIONS=1",
@@ -51,7 +51,7 @@ posix_defines = [
 linux_defines = posix_defines + [
     "_GNU_SOURCE",
     "HAVE_GETAUXVAL=1",
-    "HAVE_MALLINFO=1",
+    # "HAVE_MALLINFO=1",
     "HAVE_SBRK=1",
     "HAVE_STRUCT_STAT_ST_MTIM_TV_NSEC=1",
 ]
diff --git a/utils/bazel/llvm-project-overlay/mlir/tblgen.bzl b/utils/bazel/llvm-project-overlay/mlir/tblgen.bzl
index 89b17735e005..8044779b95c0 100644
--- a/utils/bazel/llvm-project-overlay/mlir/tblgen.bzl
+++ b/utils/bazel/llvm-project-overlay/mlir/tblgen.bzl
@@ -172,7 +172,7 @@ def _gentbl_rule_impl(ctx):
         # Make sure action_env settings are honored so the env is the same as
         # when the tool was built. Important for locating shared libraries with
         # a custom LD_LIBRARY_PATH.
-        use_default_shell_env = True,
+        # use_default_shell_env = True,
         mnemonic = "TdGenerate",
     )
 
