diff --git a/cc/private/link/finalize_link_action.bzl b/cc/private/link/finalize_link_action.bzl
index 7aeb865..56eb1af 100644
--- a/cc/private/link/finalize_link_action.bzl
+++ b/cc/private/link/finalize_link_action.bzl
@@ -230,7 +230,7 @@ def finalize_link_action(
     else:
         tool_path = _cc_common_internal.get_tool_for_action(
             feature_configuration = feature_configuration,
-            action_name = link_type.action_name,
+            action_name = action_name,
         )
 
     if link_type.linker_or_archiver == USE_ARCHIVER:
diff --git a/cc/private/link/lto_indexing_action.bzl b/cc/private/link/lto_indexing_action.bzl
index d3fc3cc..1c63a07 100644
--- a/cc/private/link/lto_indexing_action.bzl
+++ b/cc/private/link/lto_indexing_action.bzl
@@ -14,7 +14,7 @@
 """Functions that create LTO indexing action."""
 
 load("@bazel_features//:features.bzl", "bazel_features")
-load("//cc/common:cc_helper_internal.bzl", "root_relative_path")
+load("//cc/common:cc_helper_internal.bzl", "root_relative_path", artifact_category = "artifact_category_names")
 load("//cc/private:cc_internal.bzl", _cc_internal = "cc_internal")
 load("//cc/private/compile:lto_compilation_context.bzl", "get_minimized_bitcode_or_self")
 load("//cc/private/link:finalize_link_action.bzl", "finalize_link_action")
@@ -222,8 +222,12 @@ def create_lto_indexing_action(
 
     # Create artifact for the merged object file, which is an object file that is created
     # during the LTO indexing step and needs to be passed to the final link.
+    object_file_extension = _cc_internal.get_artifact_name_extension_for_category(
+        cc_toolchain,
+        artifact_category.OBJECT_FILE,
+    )
     thinlto_merged_object_file = \
-        actions.declare_shareable_artifact(root_relative_path(output) + ".lto.merged.o")
+        actions.declare_shareable_artifact(root_relative_path(output) + ".lto.merged" + object_file_extension)
 
     action_outputs = \
         ([lto_artifact.imports for lto_artifact in all_lto_artifacts if lto_artifact.index] +
diff --git a/cc/toolchains/legacy_file_group.bzl b/cc/toolchains/legacy_file_group.bzl
index 0f48e9f..398b69f 100644
--- a/cc/toolchains/legacy_file_group.bzl
+++ b/cc/toolchains/legacy_file_group.bzl
@@ -39,6 +39,9 @@ LEGACY_FILE_GROUPS = {
         Label("//cc/toolchains/actions:cpp_link_nodeps_dynamic_library"),
         Label("//cc/toolchains/actions:cpp_link_executable"),
         Label("//cc/toolchains/actions:generate_def_file"),
+        Label("//cc/toolchains/actions:lto_index_for_dynamic_library"),
+        Label("//cc/toolchains/actions:lto_index_for_executable"),
+        Label("//cc/toolchains/actions:lto_index_for_nodeps_dynamic_library"),
     ],
     "objcopy_files": [
         Label("//cc/toolchains/actions:objcopy_embed_data"),
